src/main/java/ch/uzh/ifi/ase/csccrecommender/recommender/DocumentComparison.java

```diff
@@ -8,53 +8,107 @@
 
 public class DocumentComparison {
 
+  private static final int LINE_CONTEXT_SWITCH_THRESHOLD = 25;  // 25 is picked randomly
   private final Document document;
-  private final long overallContextHammingDistance;
-  private final long lineContextHammingDistance;
-  private Integer overallContextLcsDistance = null;
-  private Integer lineContextLevenshteinDistance = null;
+  private final long overallContextHammingDistance; //Could be converted into local variable
+  private final long lineContextHammingDistance; //Could be converted into local variable
+  private Double overallContextLcsDistance = null;
+  private Double lineContextLevenshteinDistance = null;
+  private final long chosenHammingDistance;
 
   public DocumentComparison(Document document, long overallContextSimHashForProposal, long lineContextSimHashForProposal) {
     this.document = document;
     long overallContextSimHash = (long) document.getField(MethodInvocationDocumentBuilder.OVERALL_CONTEXT_SIM_HASH_FIELD).numericValue();
     long lineContextSimHash = (long) document.getField(MethodInvocationDocumentBuilder.LINE_CONTEXT_SIM_HASH_FIELD).numericValue();
     this.overallContextHammingDistance = Util.hammingDistance(overallContextSimHash, overallContextSimHashForProposal);
     this.lineContextHammingDistance = Util.hammingDistance(lineContextSimHash, lineContextSimHashForProposal);
+    //Use overall hamming distance unless hamming distance of line context exceeds predefined threshold.
+    if(this.lineContextHammingDistance > LINE_CONTEXT_SWITCH_THRESHOLD){
+      this.chosenHammingDistance = this.lineContextHammingDistance;
+    }
+    else{
+      this.chosenHammingDistance = this.overallContextHammingDistance;
+    }
   }
 
-  public int compareOverallContexts(String overallContextForProposal) {
+  public double compareOverallContexts(String overallContextForProposal) {
     if (overallContextLcsDistance != null) {
       return overallContextLcsDistance;
     }
-
-    LongestCommonSubsequence longestCommonSubsequence = new LongestCommonSubsequence();
-    overallContextLcsDistance = longestCommonSubsequence.apply(document.get(
-        MethodInvocationDocumentBuilder.OVERALL_CONTEXT_FIELD),
-        overallContextForProposal);
+    overallContextLcsDistance = getNormalizedLcs(overallContextForProposal);
+    System.out.println(overallContextLcsDistance);
```

Comment:
Don't print to console. Use a logger instead and add a description. But logging is probably not even necessary here.

Reply:
Yup, just for a quick look at the result and forgot to remove it.